Sources/MessageInputBar.swift

```diff
@@ -24,133 +24,434 @@
 
 import UIKit
 
-open class MessageInputBar: UIView, UITextViewDelegate {
-
+open class MessageInputBar: UIView {
+    
+    public enum UIStackViewPosition {
+        case left, right, bottom
+    }
+    
     // MARK: - Properties
-
-    open let inputTextView: UITextView = {
-
-        let inputTextView = InputTextView(frame: .zero)
-        inputTextView.font = UIFont.preferredFont(forTextStyle: .body)
-        inputTextView.textColor = .black
-        inputTextView.placeholder = "New Message"
-        inputTextView.backgroundColor = .white
-        inputTextView.layer.borderColor = UIColor.lightGray.cgColor
-        inputTextView.layer.borderWidth = 1.0
-        inputTextView.layer.cornerRadius = 3.0
-        inputTextView.layer.masksToBounds = true
-        inputTextView.isScrollEnabled = false
-        return inputTextView
+    
+    open weak var delegate: MessageInputBarDelegate?
+    
+    /// A background view that adds a blur effect. Shown when 'isTransparent' is set to TRUE. Hidden by default.
+    open let blurView: UIView = {
+        let blurEffect = UIBlurEffect(style: .extraLight)
+        let view = UIVisualEffectView(effect: blurEffect)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.isHidden = true
+        return view
     }()
-
-    open let sendButton: UIButton = {
-
-        let sendButton = UIButton()
-        sendButton.setTitle("Send", for: .normal)
-        sendButton.setTitleColor(.sendButtonBlue, for: .normal)
-        sendButton.setTitleColor(UIColor.sendButtonBlue.withAlphaComponent(0.3), for: .highlighted)
-        sendButton.setTitleColor(.lightGray, for: .disabled)
-        sendButton.titleLabel?.font = UIFont.preferredFont(forTextStyle: .headline)
-        sendButton.isEnabled = false
-
-        return sendButton
+    
+    /// When set to true, the blurView in the background is shown and the backgroundColor is set to .clear. Default is FALSE
+    open var isTranslucent: Bool = false {
+        didSet {
+            blurView.isHidden = !isTranslucent
+            backgroundColor = isTranslucent ? .clear : .white
+        }
```

Comment:
Same as before, can we toggle the `isOpaque` property when we switch between colors that are opaque or not?

Reply:
Could you explain what you mean by this? I thought isOpaque is only used when drawing content using `draw(_:)` method. https://developer.apple.com/documentation/uikit/uiview/1622622-isopaque

Reply:
Good to know 👍 Looks like we can leave these as is.